.github/workflows/python.yaml

```diff
@@ -303,13 +311,14 @@ jobs:
         list-files: shell
         filters: |
           python:
-            - added|modified: '**.py'  
-    - name: Check for lint
+            - added|modified: '**.py'
+    
+    - name: Check docstrings
       run: |
         if [ "${{ steps.filter.outputs.python }}" == "true" ]
```

Comment:
This is wrong for doc strings since presumably this check can also handle doc strings in extension modules which come from `.cpp` files.

Reply:
@arthurp It looks like it does not check docstring in the extension modules. I am adding the Napoleon Sphinx extension to parse Google style docstrings, but I can't find a check to see if the docstrings are correct / present.

Reply:
I also added an example

Reply:
You are only checking the python *files* here. If you wanted to check the extensions you would need to when the loaded *modules* instead. I don't know if `pydocstyle` supports that.

Reply:
This can be addressed later of course.